crates/red_knot_python_semantic/src/types/diagnostic.rs

```diff
@@ -720,9 +723,32 @@ impl<'db> TypeCheckDiagnosticsBuilder<'db> {
         );
     }
 
+    pub(super) fn add_invalid_exception_raised(&mut self, db: &dyn Db, node: &ast::Expr, ty: Type) {
+        self.add_lint(
+            &INVALID_EXCEPTION,
+            node.into(),
+            format_args!(
+                "Cannot raise object of type `{}` (must be a `BaseException` subclass or instance)",
+                ty.display(db)
+            ),
+        );
+    }
+
+    pub(super) fn add_invalid_exception_cause(&mut self, db: &dyn Db, node: &ast::Expr, ty: Type) {
```

Comment:
```suggestion
    pub(super) fn add_invalid_exception_caught(&mut self, db: &dyn Db, node: &ast::Expr, ty: Type) {
```

Reply:
`add_invalid_exception_cause` is correct here; this error is about the "cause", or the `e` in `raise Foo from e`, it's not about catching exceptions.